velox/type/TimestampUtil.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/base/Exceptions.h"
+
+namespace facebook::velox {
+
+struct TimestampToStringOptions {
+  enum Precision : int8_t {
+    kMilliseconds = 3,
+    kNanoseconds = 9,
+  } precision = kNanoseconds;
+
+  bool zeroPaddingYear = false;
+  char dateTimeSeparator = 'T';
+
+  enum Mode : int8_t {
+    kFull,
+    kDateOnly,
+    kTimeOnly,
+  } mode = kFull;
+};
+
+/// A static class that holds helper functions for TIMESTAMP type.
+class TimestampUtil {
+ public:
+  // Our own version of gmtime_r to avoid expensive calls to __tz_convert.  This
+  // might not be very significant in micro benchmark, but is causing
+  // significant context switching cost in real world queries with higher
+  // concurrency (71% of time is on __tz_convert for some queries).
+  //
+  // Return whether the epoch second can be converted to a valid std::tm.
+  static bool epochToUtc(int64_t seconds, std::tm& out);
+
+  static std::string
+  tmToString(const std::tm&, int nanos, const TimestampToStringOptions&);
```

Comment:
Would you document this method?

Make sure all public APIs use /// for comments and internal code uses //.

CC: @Yuhta